### PR TITLE
feat: add 404 page

### LINKED ISSUE #360: Add custom 404 page
Show the custom 404 page rather than the nginx default
Screen Shot 2019-09-19 at 1 59 57 PM

### PR DESCRIPTION
**Pull request by vpicone**
Closes #360 

This also makes React the first tab under get-started/develop and uses it for the suggested link.

`Now` isn't configured to point to the 404 so you have to go to `/404` to see it

### CODE REVIEW

File: src/pages/404.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { FourOhFour } from 'gatsby-theme-carbon';
+
+const links = [
+  { href: '/get-started/about-carbon', text: 'About Carbon' },
+  { href: '/components/overview', text: 'Components overview' },
+  { href: '/get-started/development', text: 'Get started: development' },
```
Comment (emyarod) [inline src/pages/404.js]: is this link still WIP?

Comment (vpicone) [reply src/pages/404.js]: What do you mean?

Comment (emyarod) [reply src/pages/404.js]: only this link is broken for me in the deploy preview so I thought it was still WIP
image

Comment (emyarod) [reply src/pages/404.js]: probably needs an `index.mdx`?

**Pull request merged**